### PR TITLE
Improvement: update DatabaseQueryException to log when uncaught

### PR DESCRIPTION
**Pull request by JasonTheAdams**
## Description

The `DatabaseQueryException` was created before the uncaught exception logging was created, as such it's extending the base `Exception`, instead of the GiveWP primitive. This PR remedies this and adds the methods to best log the broken query.

I also cleaned up a couple minor, related things.

## Affects

Specifically, how uncaught query exceptions are handled. But it does broadly touch all queries.

## Testing Instructions

Throw a `DatabaseQueryException` without catching it, then check the logs to make sure it's there.

## Pre-review Checklist

-   [ ] Acceptance criteria satisfied and marked in related issue
-   [x] Relevant `@unreleased` tags included in DocBlocks
-   [ ] Includes unit and/or end-to-end tests
-   [x] [Self Review](https://give.gitbook.io/development-manual/devops/github/code-reviews#self-review) of code and UX completed

### CODE REVIEW

**Comment by JasonTheAdams** [comment]
@alaca Mind giving this little PR a review?

File: src/Revenue/Repositories/Revenue.php [hunk 1]
```diff
@@ -5,6 +5,7 @@
 use Give\Framework\Database\DB;
 use Give\Framework\Database\Exceptions\DatabaseQueryException;
```
Comment (alaca) [inline src/Revenue/Repositories/Revenue.php:6]: There are a few unused namespaces here.

**Review by alaca — APPROVED** [review]
Looks good, much better than what we had before :+1:

File: src/Revenue/Repositories/Revenue.php (hunk 1 above)
Comment (JasonTheAdams) [reply src/Revenue/Repositories/Revenue.php]: Good catch! Cleaned up in 63d40d136

**Pull request merged**